### PR TITLE
Support lifecycle hooks when spawning single-user Notebook server.

### PR DESCRIPTION
**Pull request by ktong**
Lifecycle hook can be used to preparing or cleanup some resources (e.g. OAuth token, service accounts).

### CODE REVIEW

File: kubespawner/spawner.py [hunk 1]
```diff
@@ -444,6 +444,31 @@ def _hub_connect_port_default(self):
         """
     )
 
+    lifecycle = Dict(
```
Comment (yuvipanda) [inline kubespawner/spawner.py]: Can you rename this to singleuser_lifecycle_hooks so it's clearer what it is?

File: kubespawner/objects.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def make_pod_spec(
     cpu_guarantee,
     mem_limit,
     mem_guarantee,
+    lifecycle,
```
Comment (yuvipanda) [inline kubespawner/objects.py]: rename this to lifecycle_hooks too for clarity?

**Comment by ktong** [comment]
Names changed:)

**Comment by yuvipanda** [comment]
w00t! thank you for the patch :)

**Pull request merged**